content/learn/book/control-flow/change-detection.md

```diff
@@ -5,3 +5,170 @@ insert_anchor_links = "right"
+
+The [`Changed<T>`] query filter detects when a component has been changed. Adding a new component
+counts as "changed" - in otherwords, this is a superset of [`Added<T>`].
+
+```rust
+// Detecting added components
+fn detect_added_position(query: Query<Entity, Added<Position>>) {
+    for entity in &query {
+        println!("Entity {:?} was just given a Position component", entity);
+    }
+}
+
+// Detecting changed components.
+fn detect_changed_position(query: Query<(Entity, &Position), Changed<Position>>) {
+    for (entity, position) in &query {
+        println!("Entity {:?} position changed to {:?}", entity, position);
+    }
+}
+```
+
+Removing components works differently, see the section below.
+
+## Checking for changes
+
+In some cases, you may want to know whether or not an entity's components have changed, but at
+the same time want to access all the entities, even the ones that have _not_ changed.
+
+You can use the special [`Ref<T>`] query parameter instead of `&` for immutable access. This provides
+methods [`.is_changed()`] and [`.is_added()`] which return true if the component was changed or
+added, respectively.
+
+For mutable access, you don't need to do anything special to get the change detection methods. Using
+`&mut` in a query parameter causes Bevy to use the special [`Mut<T>`] type, which also supports change
+detection.
+
+```rust
+// Using Ref<T> for change detection with immutable access
+fn check_position_changes(query: Query<(Entity, Ref<Position>)>) {
+    for (entity, position) in &query {
+        if position.is_changed() {
+            println!("Entity {:?} position changed to {:?}", entity, position);
+        } else {
+            println!("Entity {:?} position remained {:?}", entity, position);
+        }
+    }
+}
+
+// Using Mut<T> for change detection with mutable access
+fn check_and_update_position(mut query: Query<(Entity, Mut<Position>)>) {
+    for (entity, mut position) in &mut query {
+        if position.is_changed() {
+            println!("Entity {:?} position changed to {:?} before update", entity, *position);
+            position.x += 1.0; // This will mark the component as changed
+        }
+    }
+}
+```
+
+## Resources
+
+The [`Res<T>`] and [`ResMut<T>`] provide the same [`.is_added()`] and [`.is_changed()`] methods as
+components do:
+
+```rust
+fn detect_changed_score(score: Res<Score>) {
+    if score.is_changed() {
+        println!("The score changed to {}", score.0);
+    }
+}
+```
+
+## Removed Components
+
+Change detection works differently for removed components, since the component (and possibly the
+entity) no longer exists!
+
+To detect when components are removed, you can use the [`RemovedComponents`] event:
+
+```rust
+fn detect_removed_position(mut removed: RemovedComponents<Position>) {
+    for entity in removed.iter() {
+        println!("Entity {:?} just lost its Position component", entity);
+    }
+}
+```
```

Comment:
IME using `RemovedComponents<T>` is an anti-pattern for the  majority of cases. It can miss component removals when used in `FixedUpdate`, and doesn't let you access the component values. Using `OnRemove` or the `on_remove` hook tends to be preferable, excluding specific cases where you want to quickly process a very large number of removals at once, and don't need the component values.

So I would *at least* link to `OnRemove` here, though again I don't think we have a section on observers yet...